go/vt/sqlparser/ast.go

```diff
@@ -1803,6 +1810,29 @@ func (node *OptLike) walkSubtree(visit Visit) error {
 	return Walk(visit, node.LikeTable)
 }
 
+type OptSelect struct {
+	Select SelectStatement
+	As bool
```

Comment:
This bool isn’t necessary. `As` is just syntactic sugar so it can be discarded after parsing. Under `Format` we can always display with or without the `As`. In general, the only things that we persist in structs are things that are significant and alter the meaning of the statement.